### ISSUE TITLE
Connection log improvement

### ISSUE THREAD

**Issue by McyD** [Tyrsis/SE-Community-Mod-API#38]
As it stands now the connection log is about useless.  If I am looking for, say a greifer, I can see steam ID joined which does not help.  To link the griefers steam id to there in-game name, we either have to get lucky and they created a faction, where both is logged. Or in my case scour my chat logs as I have @name in the join message and try and link the steam id join time to the players name.  Easy to do if it is a slow period of time, but impossible to do when the server is busy due to the delay from the initial login time, to the actually in-game spawn and chat message.

IF the connection log could be improved to show both steam id and in-game name that would be great.  Or a new file created, such as players, which contains both much like how minecraft server now track player names with the new UID system it uses.

**Comment by dodexahedron** [comment]
The built-in SpaceEngineers log shows the user names who connect.  Does that get you what you need?

**Comment by McyD** [comment]
I don't have these and don't believe I ever have.  Closest would be the SESE_Date_Time.log.  Which only shows the steam ID.

**Comment by Tyrsis** [comment]
Wait you don't have the dedicated logs?  These should be in the same directory as the Internal logs.  If you can locate them that would actually help with an issue people have with login tracking.  They _should_ be created in the same directory as all the other logs, but if they are being put elsewhere, that could be an issue for something out.  (Pathing in extender is horrible)

**Comment by McyD** [comment]
Nope, I have the SEModAPIInternal and SEModAPIInternal_chat logs.  There is one SpaceengineersDedicated log, if that is the one you are talking about.  However, it is not useful as it gets overwritten every restart as the date/time is not appended to the name.

I will look again once I am home from work but I have no other logs that I have been able to find.

**Comment by Tyrsis** [comment]
It does get overwritten, but old logs should be moved to SpaceEngineersDedicated_date.log before the new one is written to.

**Comment by McyD** [comment]
Mine are definitely not getting renamed, and are just getting overwritten.  Maybe this is tied into the "log file is use" bug that is crashing out restarts that started a few weeks ago.

**Comment by dodexahedron** [comment]
If you're having issues with the dedicated server logs themselves, that would actually be an issue to report to Keen, since that's entirely handled by SE itself, and it's a SESE-managed function.

There are three potential places for the logs to get dumped, depending on how you have everything set up.  They can be in any of the following places:
 * The game directory, in a folder called logs
 * ProgramData\SpaceEngineersDedicated\WorldName
 * %userprofile%\AppData\Roaming\SpaceEngineersDedicated\Worldname


What you're looking for will be a file named "SpaceEngineersDedicated_20150126_123456.log" (following the pattern of SpaceEngineersDedicated_YYYYMMDD_HHmmss.log).  If you can't find these files, open up a command prompt on your server, go to the root of the drive your world is saved on, and enter this command to try to find the logs:

dir /s /b SpaceEngineersDedicated_*.log

That will print to the screen the path of all log files starting with SpaceEngineersDedicated, so you can find them.
If that turns up nothing, seriously post your findings on the Keen forums.

**Comment by McyD** [comment]
Might have figured out the log issue.  My son just got SE, setting up a private server to tteach him the game.  Anyway, when setting up a dedicated server there is an option to "add date to log filename".  Chances are I did not notice or check this during the server setup.  I wil have to open up the SBC and see if I can manually set it.

**Comment by McyD** [comment]
Found the fix to the logs not being dated.  You have to start the vanilla dedi launcher on the server, and set it to append the date/time to the logs.  Which if like me and you set up the world elsewhere and upload the files to the server, you have most likely never ran.
